techniques/system/common/1.0/update.st

```diff
@@ -99,14 +99,11 @@ bundle common server_info
         string  => translatepath("${sys.workdir}/policy_server.dat"),
         comment => "Path to file containing address to policy server";
 
-   # Use localhost as default policy_server if no policy_server.dat exists
-    policy_server_set::
-      "cfserved" string => readfile("${policy_server_file}", 2048);
-    !policy_server_set::
-      "cfserved" string => "rudder";
-
-  classes:
-      "policy_server_set" expression => fileexists("${server_info.policy_server_file}");
+      # Use "rudder" as default policy_server if no policy_server.dat exists
+      "cfserved" string => ifelse(fileexists("${server_info.policy_server_file}"),
+                                  readfile("${policy_server_file}", 2048),
```

Comment:
i'm not sure you can safely use functions within ifelse - at least i'm pretty sure that it will fail with agent using cfengine 3.5